agent/fw_drupal.c

```diff
@@ -155,40 +227,108 @@ NR_PHP_WRAPPER(nr_drupal_http_request_exec) {
    * (nr_php_swap_user_functions), which breaks as since PHP 7.3 user
    * functions are stored in shared memory.
    */
-  bool is_drupal_7;
-  zval* arg = NULL;
-  zend_execute_data* ex = nr_get_zend_execute_data(NR_EXECUTE_ORIG_ARGS);
+  nr_drupal_http_request_add_headers(NR_EXECUTE_ORIG_ARGS);
 
-  if (NULL == ex) {
-    NR_PHP_WRAPPER_LEAVE
-  }
+  NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_DRUPAL);
 
-  is_drupal_7 = (ex->func->common.num_args == 2);
+  NRPRG(drupal_http_request_depth) += 1;
```

Comment:
Is this something we might want to flag with a comment that will be affected by fibers support?

Reply:
Will a comment `not thread/fiber safe` [here](https://github.com/misiekn/newrelic-php-agent/blob/oapi/drupal_http_request/agent/php_newrelic.h#L410) be ok?

Reply:
Sure - although on second thought maybe just the NRPRG() macro is enough of a signature for us to l look for in the future?

Reply:
I agree with this statement - NRPRG() macro is enough of a signature for us to l look for when adding fibers support. Any global state for that matter.